src/hooks/useDialog/useDialog.tsx

```diff
@@ -0,0 +1,121 @@
+import MessageDialog, { MessageDialogProps } from '@/components/Dialogs/MessageDialog'
+import { createId } from '@/utils/createId'
+import React, { useCallback, useEffect, useState, useContext, useMemo } from 'react'
+import { TransitionGroup } from 'react-transition-group'
+
+type DialogRendererProps = {
+  component: React.FunctionComponent
+}
+
+interface DialogsContainerProps {
+  dialogs: Record<string, React.FunctionComponent>
+  component?: React.ComponentType
+  container?: Element
+}
+
+const DialogsRenderer = ({ component, ...rest }: DialogRendererProps) => component(rest)
+
+export const DialogsContainer = ({ dialogs }: DialogsContainerProps) => {
```

Comment:
I guess we could just move it inside the context and get rid of the props and additional components?